test/upgrade/prober/wathola/config/structure.go

```diff
@@ -31,10 +31,16 @@ type ReceiverProgressConfig struct {
 	Duration time.Duration
 }
 
+// ReceiverErrorConfig holds error reporting config of the receiver
+type ReceiverErrorConfig struct {
```

Comment:
To make it easy to configure we need to export that setting to be available in prober configuration:

https://github.com/knative/eventing/blob/8d571d4/test/upgrade/prober/configuration.go#L86-L105

Reply:
I thought we can already do the configurations via `config.toml`. What's the additional benefit in this case?

Reply:
Yeah, but how you would influence config.toml without changing the code? I would like to have a solution that will not require me to change this upstream code, to configure behavior. Especially if that is related to failure gating.

Reply:
In knative-gcp we have: https://github.com/google/knative-gcp/tree/master/test/upgrade/prober. So I just overwrite the values in `config.toml`, otherwise we have to export virtually all the configurations. I am ok with bringing it up to the top level, but I am wondering if there is a better solution like passing the `config.toml` as an input argument to the prober. Otherwise the config.toml file will be not much different than hardcoding default values.

Reply:
Sure, you can do it. The same is for us https://github.com/openshift/knative-eventing/blob/release-next/test/upgrade/prober/config.toml but we like to not change upstream code if we can. That's by choise.

But, also from practical point of view, we use robots to sync upstream code to our forks, and then [apply patches](https://github.com/openshift/knative-eventing/tree/master/openshift/patches). Staying as close to the upstream helps to make those robots work stable.

Reply:
With #4615 would you agree that we don't have to bring up all the configurations to the prober configuration? Since after it there shouldn't be any difference between configuring prober configuration and `config.toml`. Basically the reason I am saying this is that there are several variables IMO needs to be tuned to have it working, like `Cooldown`, keep bringing those configuration to prober configuration will make it cumbersome and duplicate work for `config.toml`.

Reply:
I agree.

Reply:
Awesome, this should avoid the need to keep bringing configuration to the top level.